spec/eventception/event_handler_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'eventception'
+require 'spec_helper'
+require 'support/listener'
+require 'support/subscriber'
+require 'support/test_event'
+require 'support/listener'
+
+describe Eventception::EventHandler do
+  let(:event_handler) { described_class.new(listener: listener, method: method) }
+  let(:method) { 'on_before' }
+  let(:method2) { 'on_after' }
+  let(:listener) { Eventception::Support::Listener.new }
+  let(:listener2) { Eventception::Support::Listener.new }
+
+  describe '#call' do
+    it 'executes the listener method' do
+      expect(listener).to receive(method).and_call_original
```

Comment:
Good catch here does not make any different call the original method is just making sure it's called.